### PR TITLE
[DO-1752] Create PEP19 scoring rule

### PR DESCRIPTION
**Pull request by Andrii17s**
Rule 19
Declared ownership and/or right of use of more than 5 cars

### CODE REVIEW

File: business_register/pep_scoring/rules.py [hunk 1]
```diff
@@ -104,3 +104,33 @@ def calculate_weight(self):
...
+            ).count()
+            total = have_car + have_rights
+            if total > max_count:
+                weight = 0.4
+                data = {
+                    "declaration_id": declaration_id,
+                    "count": total,
+                }
+                return weight, data
+        return 0
```
Comment (roman-tiukh) [inline business_register/pep_scoring/rules.py:136]:
```suggestion
        return 0, {}
```

File: business_register/pep_scoring/rules.py [hunk 2]
```diff
@@ -104,3 +104,33 @@ def calculate_weight(self):
...
+            ).count()
+            have_rights = VehicleRight.objects.filter(
+                pep_id=self.pep.id,
+            ).count()
+            total = have_car + have_rights
+            if total > max_count:
+                weight = 0.4
+                data = {
+                    "declaration_id": declaration_id,
+                    "count": total,
```
Comment (roman-tiukh) [inline business_register/pep_scoring/rules.py:133]:
```suggestion
                    "total_cars_have": total,
```

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

**Review by roman-tiukh — APPROVED** [review]

File: business_register/pep_scoring/rules.py [hunk 3]
```diff
@@ -256,3 +256,39 @@ def calculate_weight(self) -> Tuple[Union[int, float], dict]:
...
+
+    rule_id = ScoringRuleEnum.PEP19
+
+    class DataSerializer(serializers.Serializer):
+        total_cars_have = serializers.IntegerField(min_value=0, required=True)
+
+    def calculate_weight(self) -> Tuple[Union[int, float], dict]:
+        year = self.declaration.year
+        max_count = 5  # max amount of cars
+        owner_types = [VehicleRight.OWNERSHIP, VehicleRight.BENEFICIAL_OWNERSHIP, VehicleRight.JOINT_OWNERSHIP,
```
Comment (IegorSoboliev) [inline business_register/pep_scoring/rules.py:277]: I think here we can sum all cars. Please, verify this with Nastya

File: business_register/pep_scoring/rules.py [hunk 4]
```diff
@@ -256,3 +256,39 @@ def calculate_weight(self) -> Tuple[Union[int, float], dict]:
...
+                       VehicleRight.COMMON_PROPERTY]
+        have_car = Vehicle.objects.filter(
+            declaration=self.declaration.id,
+        ).count()
+        have_rights = VehicleRight.objects.filter(
```
Comment (IegorSoboliev) [inline business_register/pep_scoring/rules.py:282]: Do we have several rights on the same car?

File: business_register/pep_scoring/rules.py [hunk 5]
```diff
@@ -256,3 +256,39 @@ def calculate_weight(self) -> Tuple[Union[int, float], dict]:
...
+            pep_id=self.pep.id,
+            acquisition_date__year=year,
```
Comment (IegorSoboliev) [inline business_register/pep_scoring/rules.py:284]: What is the reason to use acquisition_date__year? My car is 4-years old, and it should be counted

File: business_register/pep_scoring/rules.py [hunk 6]
```diff
@@ -256,3 +256,39 @@ def calculate_weight(self) -> Tuple[Union[int, float], dict]:
...
+    """
+    Rule 19 - PEP19
+    weight - 0.4
+    Declared ownership and/or right of use of more than 5 cars
+    """
+
+    rule_id = ScoringRuleEnum.PEP19
+
+    class DataSerializer(serializers.Serializer):
+        total_cars_have = serializers.IntegerField(min_value=0, required=True)
```
Comment (IegorSoboliev) [inline business_register/pep_scoring/rules.py:272]: There are different types of Vehicle. I think we need only cars. Please, verify this with Nastya

**Pull request closed without merging**